client/app/components/user/session.service.js

```diff
@@ -241,25 +245,114 @@ class SessionService {
             nichandle: currentUser.nichandle,
             fullName: `${currentUser.firstName} ${currentUser.lastName}`,
             subLinks: [
+                // My Account
                 {
+                    name: "user.account",
                     title: this.translator.tr("global_account"),
-                    url: "#/useraccount/infos"
-                }, {
+                    url: "#/useraccount/infos",
+                    subLinks: [{
+                        title: this.translator.tr("global_account_infos"),
+                        url: "#/useraccount/infos"
+                    }, {
+                        title: this.translator.tr("global_account_security"),
+                        url: "#/useraccount/security"
+                    },
+                        (this.constants.target === "EU" || this.constants.target === "CA") && {
```

Comment:
I'm conflicted as I like the brevity of it but I also find it quite hard to read.

Wouldn't this be clearer if it were behind a `if` statement ? Or maybe a function that adds the items only if a condition is fulfilled ?

`function addItem(title, url, condition)` ? I don't know, just a thought

Reply:
Yes, it was to avoid the combination of `if` and `push`, since we have condition for the subLinks.
Your idea is to replace it with something like
```
addItem(!currentUser.isEnterprise, {
    name: "user.billing",
    title: this.translator.tr("global_billing"),
    url: "#/billing/history",
    subLinks: [{
        title: this.translator.tr("global_billing_history"),
        url: "#/billing/history"
    }, {
        title: this.translator.tr("global_billing_payments"),
        url: "#/billing/payments"
    }]
})
```
?

Reply:
IMHO I like it that way. Won't it insert false values in the structure though?

Reply:
Yes, it's for the one-time binding in the navbar component. It keep a watcher since the value is not anymore `undefined`. The `false` value will be ignored when building the menu.